tensorflow_probability/python/experimental/mcmc/particle_filter.py

```diff
@@ -435,13 +472,410 @@ def seeded_one_step(seed_state_results, _):
+      steps=0,
+      parent_indices=smc_kernel._dummy_indices_like(init_state.log_weights),
+      incremental_log_marginal_likelihood=batch_zeros,
+      accumulated_log_marginal_likelihood=batch_zeros,
+      seed=samplers.zeros_seed())
+
+  initial_state = smc_kernel.WeightedParticles(
+      particles=(initial_state,
+                 inner_weighted_particles,
+                 initial_filter_results.parent_indices,
+                 initial_filter_results.incremental_log_marginal_likelihood,
+                 initial_filter_results.accumulated_log_marginal_likelihood),
+      log_weights=initial_log_weights,
+      extra=(tf.constant(0),
+             initial_filter_results.seed)
+  )
+
+  outer_propose_and_update_log_weights_fn = (
+      _outer_particle_filter_propose_and_update_log_weights_fn(
+          outer_rejuvenation_criterion_fn=outer_rejuvenation_criterion_fn,
+          inner_observations=inner_observations,
+          inner_transition_fn=inner_transition_fn,
+          inner_proposal_fn=inner_proposal_fn,
+          inner_observation_fn=inner_observation_fn,
+          inner_resample_fn=inner_resample_fn,
+          inner_resample_criterion_fn=inner_resample_criterion_fn,
+          parameter_proposal_kernel=parameter_proposal_kernel,
+          initial_parameter_prior=initial_parameter_prior,
+          num_transitions_per_observation=num_transitions_per_observation,
+          unbiased_gradients=unbiased_gradients,
+          inner_initial_state_prior=inner_initial_state_prior,
+          inner_initial_state_proposal=inner_initial_state_proposal,
+          num_inner_particles=num_inner_particles,
+          num_outer_particles=num_outer_particles,
+          extra_fn=extra_fn
+      )
+  )
+
+  traced_results = sequential_monte_carlo(
+      initial_weighted_particles=initial_state,
+      propose_and_update_log_weights_fn=
+      outer_propose_and_update_log_weights_fn,
+      resample_fn=outer_resample_fn,
+      resample_criterion_fn=outer_resample_criterion_fn,
+      trace_criterion_fn=outer_trace_criterion_fn,
+      static_trace_allocation_size=static_trace_allocation_size,
+      parallel_iterations=parallel_iterations,
+      unbiased_gradients=unbiased_gradients,
+      num_steps=num_timesteps,
+      particles_dim=0,
+      trace_fn=outer_trace_fn,
+      seed=loop_seed
+  )
+
+  return traced_results
+
+
+def _outer_particle_filter_propose_and_update_log_weights_fn(
+        inner_observations,
+        inner_transition_fn,
+        inner_proposal_fn,
+        inner_observation_fn,
+        initial_parameter_prior,
+        inner_initial_state_prior,
+        inner_initial_state_proposal,
+        num_transitions_per_observation,
+        inner_resample_fn,
+        inner_resample_criterion_fn,
+        outer_rejuvenation_criterion_fn,
+        unbiased_gradients,
+        parameter_proposal_kernel,
+        num_inner_particles,
+        num_outer_particles,
+        extra_fn
```

Comment:
I think we can remove this `extra_fn` argument, too.

For now, SMC^2 can just use the `extra` field of the outer `WeightedParticles` for storing for storing the random seed of the inner filter results.